ez-launch/sources/ez_launch.move

```diff
@@ -0,0 +1,413 @@
+        let object_signer_address = signer::address_of(&object_signer);
+        let collection_details_obj = borrow_global_mut<CollectionDetails>(object_signer_address);
+
+        vector::length(&collection_details_obj.to_be_minted) == 0
+    }
+
+    // ================================= Helper  ================================= //
+
+    fun create_collection_internal(
+        creator: &signer,
+        description: String,
+        name: String,
+        uri: String,
+        mutable_collection_metadata: bool,
+        mutable_token_metadata: bool,
+        random_mint: bool,
+        is_soulbound: bool,
+        tokens_burnable_by_creator: bool,
+        tokens_transferable_by_creator: bool,
+        max_supply: Option<u64>,
+        mint_fee: Option<u64>,
+        royalty_numerator: Option<u64>,
+        royalty_denominator: Option<u64>,
+    ): Object<Collection> acquires CollectionDetails {
+        let creator_addr = signer::address_of(creator);
+
+        assert!(option::is_some(&royalty_numerator) == option::is_some(&royalty_denominator), error::invalid_argument(EROYALTY_CONFIGURATION_INVALID));
+        let royalty = if (option::is_some(&royalty_numerator) && option::is_some(&royalty_denominator)) {
+            option::some(royalty::create(option::extract(&mut royalty_numerator), option::extract(&mut royalty_denominator), creator_addr))
+        } else {
+            option::none()
+        };
+        let constructor_ref = if (option::is_some(&max_supply)) {
+            collection::create_fixed_collection(
+                creator,
+                description,
+                option::extract(&mut max_supply),
+                name,
+                royalty,
+                uri,
+            )
+        } else {
+            collection::create_unlimited_collection(
+                creator,
+                description,
+                name,
+                royalty,
+                uri,
+            )
+        };
+
+        collection_components::create_refs(
+            &constructor_ref,
+            mutable_collection_metadata, // mutable_description
+            mutable_collection_metadata, // mutable_uri
+            mutable_collection_metadata, // mutable_royalty
+        );
+
+        let collection_properties = collection_components::create_properties(
+            mutable_collection_metadata, // mutable_description
+            mutable_collection_metadata, // mutable_uri
+            mutable_token_metadata, // mutable_token_description
+            mutable_token_metadata, // mutable_token_name
+            mutable_token_metadata, // mutable_token_properties
+            mutable_token_metadata, // mutable_token_uri
+            mutable_collection_metadata, // mutable_royalty
+            tokens_burnable_by_creator,
+            tokens_transferable_by_creator,
+        );
+        collection_components::init_collection_properties(&constructor_ref, collection_properties);
```

Comment:
We can replace all this with 1 line in this PR: https://github.com/aptos-labs/token-minter/pull/32

:)